cmd/influx_inspect/buildtsi/util.go

```diff
@@ -0,0 +1,218 @@
+package buildtsi
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/influxdata/influxdb/models"
+	"github.com/influxdata/influxdb/tsdb"
+	"github.com/influxdata/influxdb/tsdb/engine/tsm1"
+	"github.com/influxdata/influxdb/tsdb/index/tsi1"
+	"go.uber.org/zap"
+)
+
+func IndexShard(sfile *tsdb.SeriesFile, dataDir, walDir string, maxLogFileSize int64, batchSize int, log *zap.Logger, verboseLogging bool) error {
+	log.Info("Rebuilding shard")
+
+	// Check if shard already has a TSI index.
+	indexPath := filepath.Join(dataDir, "index")
+	log.Info("Checking index path", zap.String("path", indexPath))
```

Comment:
Maybe we could do a `log = log.With(zap.String("path", indexPath)` here?

Reply:
^ I think adding `path` to the logger might break things further down, so maybe keep it as it is 😄